### PR TITLE
Data Integrity During On-Boarding - Best Practices

### PR DESCRIPTION
**Pull request by mrjones-plip**

### CODE REVIEW

File: content/en/apps/guides/data/on-boarding-data-integrity.md
Comment (mrjones-plip) [inline content/en/apps/guides/data/on-boarding-data-integrity.md:125]: @Marina-Kenf - thanks for your input to answer @binokaryg question on this in the draft of this document. How does this look?

```suggestion
7. Change the fields by deleting "_rev" field and also "_attachments" if uploading in a different instance. This can be carefully done by manually editing the JSON locally before uploading or by running `DELETE` calls against the CouchDB database.
```

**Comment by mrjones-plip** [comment]
@abbyad  - I struggled as to where to add this on the docs site - open to moving it if you think it ended up in the wrong spot!

File: content/en/apps/guides/data/on-boarding-data-integrity.md
Comment (Marina-Kenf) [reply content/en/apps/guides/data/on-boarding-data-integrity.md:125]: @mrjones-plip, I'll rather remove the deletion part in CouchDB unless we want to destroy the original documents on the initial instance.

Comment (mrjones-plip) [reply content/en/apps/guides/data/on-boarding-data-integrity.md:125]: Awesome - thanks @Marina-Kenf ! I've updated the text and will commit the change.  

@binokaryg - lemme know if this satisfies your original "how" question!

File: content/en/apps/guides/data/on-boarding-data-integrity.md
Comment (abbyad) [inline content/en/apps/guides/data/on-boarding-data-integrity.md:28]: It would be nice to avoid having medicmobile.org in here, so perhaps `project.dev.example.com`

File: content/en/apps/guides/data/on-boarding-data-integrity.md
Comment (abbyad) [inline content/en/apps/guides/data/on-boarding-data-integrity.md:30]: Although the informal tone is appropriate for a guide, the expressions like `is here` and `click this` make it hard to understand. Also, it's better to [avoid latin word/phrases](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#latin-phrases) like `NB` and even `caveat`, and use a [note box](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#notes-and-tips).

```suggestion
{{% alert title="Note" %}}Careful when selecting **About** to not select **Logout**. If accidental log outs are a frequent problem the logout option can be removed [from the menu by setting `can_log_out_on_android` to false]({{< relref "apps/reference/app-settings/user-permissions" >}}).{{% /alert %}}
```

File: content/en/apps/guides/data/on-boarding-data-integrity.md
Comment (abbyad) [inline content/en/apps/guides/data/on-boarding-data-integrity.md:16]: I keep waiting for part two of that first sentence because of the "important to both...."

File: content/en/apps/guides/data/on-boarding-data-integrity.md
Comment (abbyad) [inline content/en/apps/guides/data/on-boarding-data-integrity.md:12]: This block may be better as a note, but we can revisit after the main structure/overview is addressed.

**Review by abbyad — CHANGES_REQUESTED** [review]
Nice to see this coming together!

When reading the doc I feel as though I am reading a troubleshooting guide, but missing the instructions. I found myself asking _How do we recommend CHWs be trained?_, which is important context for this guide. Can that be included before discussing the ways it can go wrong and remediation, and make this a broader _"Best practices for onboarding users"_ guide? If it is a single page, then that would mean moving everything over to something like `guides/onboarding`, but could also be split into two guides so that the remediation stands along in `guides/data`. Either way, I think that having that intro will help with the overall structure and flow for this content.

I will review the text more closely on the next pass, but left some comments that could be addressed now. Also, as a general note there seems to be a tendency to use `onboarding`, rather than `on-boarding`, so would use `onboard` throughout. That would also be consistent with [our use of it in the docs site](https://docs.communityhealthtoolkit.org/apps/examples/learning-care/).

**Comment by kennsippell** [comment]
Is this assuming that each app has two APKs: a production APK and a training APK? In practice, I've seen deployments using the [medic app](https://play.google.com/store/apps/details?id=org.medicmobile.webapp.mobile&hl=en_CA&gl=US) on playstore for the training app.  then uninstalling the medic app after the training is completed.

**Comment by mrjones-plip** [comment]
>  When reading the doc I feel as though I am reading a troubleshooting guide, but missing the instructions. I found myself asking How do we recommend CHWs be trained?, which is important context for this guide. Can that be included before discussing the ways it can go wrong and remediation, and make this a broader "Best practices for onboarding users" guide? If it is a single page, then that would mean moving everything over to something like guides/onboarding, but could also be split into two guides so that the remediation stands along in guides/data. Either way, I think that having that intro will help with the overall structure and flow for this content.

I'm very open to blocking the release of this page based on missing, broader "Best practices for onboarding users" content! However, such a guide is not one I'd feel comfortable writing, as I've never been through an onboarding, let alone to the field. I would be very open to curating this content by working with teammates who have the expertise!  

Maybe we could just add single sentence (or two!) to get this PR through and open another ticket to extend it to be more broad later? I'll try my hand, but am still cool to nix my sentences for a more proper solution per above.

File: content/en/apps/guides/data/on-boarding-data-integrity.md
Comment (mrjones-plip) [reply content/en/apps/guides/data/on-boarding-data-integrity.md]: agreed! fixed in 3d1f96b .

File: content/en/apps/guides/data/on-boarding-data-integrity.md
Comment (mrjones-plip) [reply content/en/apps/guides/data/on-boarding-data-integrity.md]: that sentence is too long as it is!! removed the "both" in 7fe4410

**Comment by mrjones-plip** [comment]
> Is this assuming that each app has two APKs: a production APK and a training APK? In practice, i'm unsure which projects actually do that. I've seen deployments using the [medic app](https://play.google.com/store/apps/details?id=org.medicmobile.webapp.mobile&hl=en_CA&gl=US) on playstore for the training app. then uninstalling the medic app after the training is completed.

Right - the scenario where the training app has a different icon for the launcher and for the login, I was thinking it would be two different APKs and not using the [unbranded app](https://play.google.com/store/apps/details?id=org.medicmobile.webapp.mobile&hl=en_CA&gl=US) .  That said, I think using the unbranded app is fine alternative in a pinch!  The draw back would be that if you log out, you now need your username, password AND the URL for the dev instance.

do you think we should add some clarifying content around this?

**Comment by mrjones-plip** [comment]
> as a general note there seems to be a tendency to use onboarding, rather than on-boarding, so would use onboard throughout. That would also be consistent with [our use of it in the docs site](https://docs.communityhealthtoolkit.org/apps/examples/learning-care/).

Thanks @abbyad! fixed in f7190cc

**Comment by mrjones-plip** [comment]
> Maybe we could just add single sentence (or two!) to get this PR through and open another ticket to extend it to be more broad later? I'll try my hand, but am still cool to nix my sentences for a more proper solution per above.

@abbyad  - I gave it a try! Kinda feels like it's not enough...but maybe you think 32679fa is totally awesome!?

**Comment by mrjones-plip** [comment]
> Ah - this just got edited, thanks for the extra info! I'll report back with a commit hash so you can see what I added. Thanks for the input!

@kennsippell  - I've added more info in 7372828 - please let me know what you think!

File: content/en/apps/guides/onboarding/_index.md
Comment (mrjones-plip) [inline content/en/apps/guides/onboarding/_index.md:23]: minor tweak: we should prepend three spaces so this renders as part of the above bullet point:
```suggestion
   ![CHT with border](border.png)
```

File: content/en/apps/guides/onboarding/_index.md
Comment (mrjones-plip) [inline content/en/apps/guides/onboarding/_index.md:16]: We should deep link to the technical how to and I think use "flavor" over "brand" to stick with existing vernacular:

```suggestion
A separate Android App can be created for training, which would point to a CHT instance dedicated to training. The training instance should have the same configuration as the production instance, and have users created for training. To differentiate the Android app used for training from the production one, create a duplicate of [your flavor]({{< relref "core/guides/android/branding" >}}) and modify the following aspects
```

File: content/en/apps/guides/onboarding/_index.md
Comment (abbyad) [reply content/en/apps/guides/onboarding/_index.md:23]: Good eye!

File: content/en/apps/guides/onboarding/_index.md
Comment (abbyad) [reply content/en/apps/guides/onboarding/_index.md:12]:
```suggestion
{{% alert title="Note" %}}The suggestions in this guide should be assessed and adapted as needed to benefit a deployment. It is important that users don't accidentally use the wrong app. The [troubleshooting guide]({{< relref "apps/guides/data/fixing-data-in-wrong-location" >}}) can help to monitor and remediate training data being in the production instance, or the opposite.{{% /alert %}}
```

File: content/en/apps/guides/onboarding/_index.md
Comment (abbyad) [reply content/en/apps/guides/onboarding/_index.md:34]:
```suggestion
It is preferable to remove the training app from devices, and [monitor the training instance for unexpected activity](< relref "apps/guides/data/fixing-data-in-wrong-location" >}}) that can be brought over to the production instance if needed.
```

File: content/en/apps/guides/onboarding/_index.md
Comment (abbyad) [reply content/en/apps/guides/onboarding/_index.md:21]:
```suggestion
- **Border & Message**: Consider adding a distinctive border and message when using the training app. This might be done in a more automated way in future versions of CHT Android, and in the meantime can be [modeled after this example](https://github.com/medic/cht-android/commit/8d077ed08dc3889ef1f4e3bad7231931bca55d87).
```

**Review by abbyad — APPROVED** [review]
This came together really nicely, thanks @mrjones-plip for pushing this through!

**Comment by abbyad** [comment]
Just noticed that the paths need updating, along with header alignment. We should update those before merging.

File: content/en/apps/guides/onboarding/_index.md
Comment (abbyad) [inline content/en/apps/guides/onboarding/_index.md:14]:
```suggestion
## Setting up a training app
```

File: content/en/apps/guides/data/fixing-data-in-wrong-location.md
Comment (abbyad) [inline content/en/apps/guides/data/fixing-data-in-wrong-location.md:84]: Path is broken. File should be moved accordingly.

**Comment by mrjones-plip** [comment]
@kennsippell  - Imma push this out, but feel free to open a ticket for me to fix anything upon my return!

Thanks @abbyad  for the great work session today, much obliged.

**Pull request merged**
